blockchain/event_filter.go

```diff
@@ -0,0 +1,198 @@
+package blockchain
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/db"
+)
+
+var errChunkSizeReached = errors.New("chunk size reached")
+
+type EventFilter struct {
+	txn             db.Transaction
+	fromBlock       uint64
+	toBlock         uint64
+	contractAddress *felt.Felt
+	keys            []*felt.Felt
+}
+
+type FilterRangeEnd byte
+
+const (
+	From FilterRangeEnd = iota
+	To
+)
+
+func newEventFilter(txn db.Transaction, contractAddress *felt.Felt, keys []*felt.Felt, fromBlock, toBlock uint64) *EventFilter {
```

Comment:
We should ensure that contract address is always non-nil

Reply:
No, it is optional